scan.go

```diff
@@ -259,16 +279,24 @@ func (s *Scan) processResponse(response pb.Message) []*ResultRow {
 		nextRegion = false
 	}
 
+	var err error
 	if nextRegion {
+		err = s.closeScan(s.server, s.location, s.id)
```

Comment:
I think you need put this line below `s.nextStartKey = s.location.EndKey`, if closeScan failed, `nextStartKey` will not be assigned correctly.